boa/src/builtins/object/gcobject.rs

```diff
@@ -152,18 +160,30 @@ impl GcObject {
                         ctx.realm.environment.push(local_env);
```

Comment:
I would move this to the bottom when we call the function, since its easier to follow the code when its in one place.